### PR TITLE
cloudformation: allow template_url parameter

### PR DESCRIPTION
**Pull request by Rob-Johnson**
Cloudformation offers a template_url parameter in the CreateStack API, allowing one to specify a URL to a Cloudfomation template in an S3 bucket to be used, rather than passing the contents of the Cloudformation stack via the template_body parameter.

This PR implements this functionality.

### CODE REVIEW

**Comment by Rob-Johnson** [comment]
rebased against devel. @jimi-c is there anything I can do to get this merged?

**Comment by geowa4** [comment]
:+1: as I'm blocked by this.

**Comment by alexanderdean** [comment]
+1 would be great to get this in!

**Comment by juliedavila** [comment]
@Rob-Johnson, sorry for the delay, this looks great, but just one thing. Can you please add a check to see if both `template` AND `template_url` are specified and subsequently error out? Alternatively, you could add in a `mutually_exclusive` param to the argument spec if you prefer.

**Comment by Rob-Johnson** [comment]
Hi @defionscode,

The current behaviour follows that of [boto](https://boto.readthedocs.org/en/latest/ref/cloudformation.html#module-boto.cloudformation.stack#create_stack):

> Conditional: You must pass TemplateURL or TemplateBody. If both are
> passed, only TemplateBody is used.

Though I agree that your suggestion is probably what _should_ happen, I think it would be surprising if the module deviated from the behaviour of boto.

What do you think?

**Comment by juliedavila** [comment]
@Rob-Johnson, even if boto supports/does it, I don't think that should translate to the module necessarily needing to support it. There can't possibly be a use-case where someone would intentionally seek out to supply both parameters for the purposes of using `TemplateBody` only. 

A more likely scenario would be someone who is refactoring their role/playbook and forgets to remove one or the other which could then lead to unintended results.

**Comment by Rob-Johnson** [comment]
Hi @defionscode,

I've rebased against devel and updated. I've had to make a couple of changes to account for upstream commits to the module, specifically https://github.com/ansible/ansible-modules-core/commit/d27df0cdb4868ddea8956ab03ce398edc534017a, but otherwise I've just added the `mutually_exclusive` option.

File: cloud/amazon/cloudformation.py [hunk 1]
```diff
@@ -76,6 +76,11 @@
     default: null
     aliases: ['aws_region', 'ec2_region']
     version_added: "1.5"
+  template_url:
+    description:
+      - Location of file containing the template body. The URL must point to a template (max size: 307,200 bytes) located in an S3 bucket in the same region as the stack. This parameter is mutually exclusive with 'template'. Either one of them is required if "state" parameter is "present"
+    required: false
+    version_added: "1.8"
```
Comment (juliedavila) [inline cloud/amazon/cloudformation.py]: Could you change this to 2.0?

Comment (Rob-Johnson) [inline cloud/amazon/cloudformation.py]: Done.

**Comment by juliedavila** [comment]
Good to go. Recommending a merge.

**Comment by Rob-Johnson** [comment]
Thanks @defionscode @bcoca

**Pull request merged**